src/Pages/RegisterForm/RegisterForm.tsx

```diff
@@ -0,0 +1,162 @@
+}
+
+function toArr(
+  str: string | number | (string | number)[]
+): (string | number)[] {
+  return Array.isArray(str) ? str : [str];
+}
+
+const MyFormItemGroup: React.FC<
+  React.PropsWithChildren<MyFormItemGroupProps>
+> = ({ prefix, children }) => {
+  const prefixPath = React.useContext(MyFormItemContext);
+  const concatPath = React.useMemo(
+    () => [...prefixPath, ...toArr(prefix)],
+    [prefixPath, prefix]
+  );
+
+  return (
+    <MyFormItemContext.Provider value={concatPath}>
+      {children}
+    </MyFormItemContext.Provider>
+  );
+};
+
+const MyFormItem = ({ name, ...props }: FormItemProps) => {
+  const prefixPath = React.useContext(MyFormItemContext);
+  const concatName =
+    name !== undefined ? [...prefixPath, ...toArr(name)] : undefined;
+
+  return <Form.Item name={concatName} {...props} />;
+};
+
+const App: React.FC = () => {
+  const [form] = Form.useForm();
+
+  const [formData, setFormData] = useState({
+    firstName: '',
+    businessEmail: '',
+    companyName: '',
+  });
+  const [errorName, setErrorName] = useState<string | null>(null);
+  const [errorEmail, setErrorEmail] = useState<string | null>(null);
+  const [errorCompany, setErrorCompany] = useState<string | null>(null);
+
+  const validateForm = () => {
+    let valid = true;
+
+ 
+    if (!/^[\p{L}\d\s._-]+$/u.test(formData.firstName)) {
+      setErrorName('Please enter only letters and the field cannot be empty');
+      valid = false;
+    } else {
+      setErrorName(null);
+    }
+
+
+    if (
+      !/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(
+        formData.businessEmail
+      )
+    ) {
+      setErrorEmail('Enter correct e-mail');
+      valid = false;
+    } else {
+      setErrorEmail(null);
+    }
+
+ 
+    if (!/^[\p{L}\d\s._-]+$/u.test(formData.companyName)) {
+      setErrorCompany('Enter correct company name');
+      valid = false;
+    } else {
+      setErrorCompany(null);
+    }
+
+    return valid;
+  };
+
+  const onFinish = () => {
+    if (validateForm()) {
+      console.log('Form submitted:', formData);
+    
+
+      form.resetFields();
+    } else {
+      console.log('Form contains errors.');
+    }
+  };
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    field: string
+  ) => {
+    const { value } = event.target;
+    setFormData(prevData => ({
+      ...prevData,
+      [field]: value,
+    }));
+  };
+
+  return (
+    <div className="form">
+      <div className="contact_us">
+        <span className="contact_us_text">Contact</span>
+        <span className="contact_us_text">us</span>
+      </div>
+      <Form
+        form={form}
+        name="form_item_path"
+        layout="vertical"
+        onFinish={onFinish}
+      >
+        <MyFormItemGroup prefix={['user']}>
+          <MyFormItemGroup prefix={['name']}>
+            <MyFormItem name="firstName" label="Name" required>
+              <Input
+                placeholder="Olivia Smith"
```

Comment:
when i place cursor in the field, I still see Olivia Smith, and I dont see the cursor is placed there, so it looks like the form is already filled and i cannot fill it in.

Check out the functionality on UI please

Reply:
Fixed. Cursor is visible